src/Magnum/Audio/Test/BufferALTest.cpp

```diff
@@ -35,18 +39,109 @@ struct BufferALTest: TestSuite::Tester {
 
     void construct();
 
+    void setData();
+    void setLoopPoints();
+    void setLoopSince();
+    void setLoopUntil();
+    void resetLoopPoints();
+
     Context _context;
 };
 
 BufferALTest::BufferALTest() {
-    addTests({&BufferALTest::construct});
+    addTests({&BufferALTest::construct,
+
+              &BufferALTest::setData,
+              &BufferALTest::setLoopPoints,
+              &BufferALTest::setLoopSince,
+              &BufferALTest::setLoopUntil,
+              &BufferALTest::resetLoopPoints});
 }
 
 void BufferALTest::construct() {
     Buffer buf;
     CORRADE_VERIFY(buf.id() != 0);
 }
 
+void BufferALTest::setData() {
+    Buffer buf;
+    constexpr char data[] { 25, 17, 24, 122, 67, 24, 48, 96 };
+
+    buf.setData(BufferFormat::Mono8, data, 22050);
+
+    ALint bufSize;
+    alGetBufferi(buf.id(), AL_SIZE, &bufSize);
+    CORRADE_VERIFY(bufSize == 8);
+}
+
+void BufferALTest::setLoopPoints() {
+    if(!_context.isExtensionSupported<Audio::Extensions::AL::SOFT::loop_points>()) {
+        CORRADE_SKIP("Extension AL_SOFT_loop_points isn't supported on this system.");
+    }
+
+    Buffer buf;
+    constexpr char data[] { 25, 17, 24, 122, 67, 24, 48, 96 };
+
+    buf.setData(BufferFormat::Mono8, data, 22050).setLoopPoints(1, 6);
+
+    Containers::Array<ALint> points{Containers::InPlaceInit, { 0, 0 }};
+    alGetBufferiv(buf.id(), AL_LOOP_POINTS_SOFT, points.data());
```

Comment:
I'm wondering, wouldn't it make sense to add a getter for these? Returning `std::pair<ALint, ALint>`, for example. I think it could make testing and debugging easier. (Then you also don't need to do the complex container testing, just `CORRADE_COMPARE(buf.loopPoints(), std::make_pair(0, 8))` for example.)

Reply:
It'd make sense. I'll add it.

Reply:
A quick test on my end leads to a success. Marking this as resolved.

Reply:
Yes, they should be, `std::pair` is a plain struct. I just realized that maybe `Range1Di` could fit the criteria here as well. Either is okay from my side.

Reply:
`std::pair` it is, then. I saw the comment a bit late. 😅